docs/user/upgrade.rst

```diff
@@ -90,6 +90,16 @@ Compatibility Matrix
 | Dependencies | N     | N     | Y     |
 +--------------+-------+-------+-------+
 
+Version 2.3.0 Upgrade Guide
++++++++++++++++++++++++++++
+
+Internal Metadata Format Change
+-------------------------------
+
+The first time a schema is accessed with a GeoMesa 2.3.x data store, the metadata will be transparently rewritten.
+After this, the schema will no longer be readable by older versions of GeoMesa. The old metadata values are
+backed up in the catalog table, in case a rollback is required.
```

Comment:
Interesting.  Is this necessary?

Reply:
Ok, I think we'll have situations where folks may want/need to read a table with multiple versions of GeoMesa.  

Either I'm misunderstanding or this would completely prevent someone from doing A/B testing like that.  How hard would it be to use a different metadata namespace or something similar?

Reply:
reading with different versions of geomesa is generally not going to work perfectly, due to the distributed runtime jar, and we discourage it (although certain queries *may* work). did you have something concrete in mind, or are you just speculating? If someone really wanted to test the same data with two different versions, you could always clone the metadata table and point it at the same set of index tables, but I don't think that's necessarily a standard use case we need to be supporting. And going forward, any schema created with 2.3+ is not going to be readable with 2.2 or earlier anyway.

Reply:
@jahhulbert-ccri has noted that one can configure different Accumulo contexts (with separate classpaths) at scan time.  It is with that use case in mind that I'm nervous having tables which once read by a newer version cannot be read with a previous version.

Reply:
that seems like a terrible idea haha. but as I said, you could do that if you cloned the metadata table before upgrading.

Reply:
scanning separate classpaths would have to be functionality that we added, right? so seems like a moot point anyway, as it's not in any older versions (or newer ones yet)